pkg/apis/build/v1alpha1/buildrun_types.go

```diff
@@ -46,18 +46,18 @@ type BuildRunSpec struct {
 	// +optional
 	Output *Image `json:"output,omitempty"`
 
-	// Used for canceling a buildrun (and maybe more later on).
+	// StateOverride is used for canceling a buildrun (and maybe more later on).
 	// +optional
-	Status BuildRunSpecStatus `json:"status,omitempty"`
+	StateOverride BuildRunStateOverride `json:"stateOverride,omitempty"`
 }
 
-// BuildRunSpecStatus defines the buildrun spec status the user can provide
-type BuildRunSpecStatus string
+// BuildRunStateOverride defines the buildrun status the user can provide to override whatever is the current status.
+type BuildRunStateOverride string
 
 const (
-	// BuildRunSpecStatusCanceled indicates that the user wants to cancel the BuildRun,
+	// BuildRunStateOverrideCanceled indicates that the user wants to cancel the BuildRun,
 	// if not already canceled or terminated
-	BuildRunSpecStatusCanceled = "BuildRunCanceled"
+	BuildRunStateOverrideCanceled = "BuildRunCanceled"
```

Comment:
And then this would be `BuildRunStateCancel` since you are requesting to cancel the build run.

Reply:
Going with `BuildRunStateCancel`